src/pyhf/modifiers/lumi.py

```diff
@@ -56,17 +61,14 @@ def apply(self, pars):
         '''
         if not self.param_viewer.index_selection:
             return
+
         tensorlib, _ = get_backend()
         if self.batch_size is None:
-            batched_pars = tensorlib.reshape(pars, (1,) + tensorlib.shape(pars))
+            lumis = self.param_viewer.get(pars)
```

Comment:
This is repeated in either case, so let's just move it before the `ifelse` block.